lib/api/objectCopy.js

```diff
@@ -280,7 +286,9 @@ function objectCopy(authInfo, request, sourceBucket,
             // skip if source and dest the same or 0 byte object
             // still send along serverSideEncryption info so algo
             // and masterKeyId stored properly in metadata
-            if (sourceIsDestination || dataLocator.length === 0) {
+            if ((sourceIsDestination && !storeMetadataParams.metaHeaders
+            ['x-amz-meta-scal-location-constraint'])
```

Comment:
also, isn't just checking whether this header is there not enough? shouldn't the check be that the specified location (if specified) is different from the data location of the source?  They could specify the location header but specify it to the same exact place.